lib/parlour/type_parser.rb

```diff
@@ -749,9 +749,10 @@ def parse_node_to_type(node)
               parse_node_to_type(args[0]), parse_node_to_type(args[1])
             )
           else
-            # TODO
-            warning "user-defined generic types not implemented, treating #{names.last} as untyped", node
-            return Types::Untyped.new
+            return Types::Generic.new(
+              names.join('::'),
```

Comment:
I *think* this should work; this is the way raw identifiers are constructed. Can you think of any cases where this would break? Maybe sending it back through `parse_node_to_type` would be better to handle any edge cases?
```suggestion
              parse_node_to_type(names.join('::')),
```

Reply:
I think this should be OK, if it's been fine for `Raw` so far.
I'm not sure how this behaves if types relative to the root `::Like::This` are used, but that would be a separate issue since `Raw` already does it